### PR TITLE
Add promotion rules article for Solidus admins

### PR DESCRIPTION
**Pull request by benjaminwil**
This is a small contribution that adds information around promotion rule types and eligibility. This is for Solidus admins.

**Checklist:**
- [x] I have followed [Pull Request guidelines](https://github.com/solidusio/solidus/blob/master/CONTRIBUTING.md#pull-request-guidelines)
- [x] I have added a detailed description into each commit message
- [x] I have updated Guides and README accordingly to this change (if needed)
- [ ] ~I have added tests to cover this change (if needed)~

### CODE REVIEW

File: guides/data/nav/users.yml [hunk 1]
```diff
@@ -21,12 +21,12 @@
...
       - title: "Overview"
         href: "/users/promotions/overview.html"
+      - title: "Promotion Rules"
+        href: "/users/promotions/promotion-rules.html"
       - title: "Promotion Actions"
         href: "/users/promotions/promotion-actions.html"
       - title: "Promotion Calculators"
         href: "/users/promotions/promotion-calculators.html"
-      - title: "Promotion Rules"
-        href: "/users/promotions/promotion-rules.html"
```
Comment (aitbw) [inline guides/data/nav/users.yml]: Is this change really necessary for this PR? I'm assuming most users would expect to have an alphabetically-sorted menu, what do you think?

File: guides/source/users/promotions/promotion-rules.html.md [hunk 2]
```diff
@@ -1 +1,50 @@
...
+
+## Available promotion rule types
+
+- **First Order**: Eligible for a customer's first order only.
+- **First Repeat Purchase Since**: Eligible for a customer's first repeat
+  purchase since a specified date.
+- **Nth Order**: Eligible for a customer's *n*th order only.
+- **Item Total**: Eligible if the order total (before any adjustments) is less
+  than or greater than a specified amount.
+- **One Use Per User**: Eligible for use one time for each user.
```
Comment (aitbw) [inline guides/source/users/promotions/promotion-rules.html.md]: I think `Eligible for use one time per user` reads better, what do you think?

**Review by kennyadsl — APPROVED** [review]
Thanks Ben!

File: guides/source/users/promotions/promotion-rules.html.md [hunk 3]
```diff
@@ -1 +1,50 @@
...
+
+## Rule matching
+
+There are two types of rule matching in Solidus' promotion system: **Match all
+of these rules** or **Match any of these rules**.
+
+- **Match all of these rules**: This setting ensure that every rule must be
+  matched before the promotion is applied. This is great if you only want to
+  apply promotions for a very specific kind of order, like for first-time
+  customers who order more than $100 USD of the T-Shirt or Tote Bag products.
```
Comment (aitbw) [inline guides/source/users/promotions/promotion-rules.html.md]: I think `(...) whose order is valued at $100 USD or more` reads better and it's more concise, what do you think?

File: guides/source/users/promotions/promotion-rules.html.md [hunk 4]
```diff
@@ -1 +1,50 @@
...
+- **Match any of these rules**: This setting allows you to create more flexible
+  promotions. For example, if you wanted to give a discount to customers who
+  buy more than $100 USD of products or are a first-time customer.
```
Comment (aitbw) [inline guides/source/users/promotions/promotion-rules.html.md]: I think it's better to use other examples in here, to avoid any possible confusions; what do you think?

**Review by aitbw — COMMENTED** [review]
Hey @benjaminwil, thanks for showing the docs some love! :heart:

I've left you some comments I'd like to discuss first before giving you my approval, let me know what you think!

**Review by jacobherrington — APPROVED** [review]
Thanks Ben! I think @aitbw has great feedback, but I'm happy with this either way. 🥇

File: guides/data/nav/users.yml (hunk 1 above)
Comment (benjaminwil) [reply guides/data/nav/users.yml]: I would expect administrators to read about "Promotion Rules" before they read about "Promotion Actions." Perhaps my commit message should have stated that more explicitly.

While the guides may not be read linearly, we can prioritize articles by putting them up higher in the menus. We do this with "Overview" articles.

This is a low-risk change so I will include it. We can revert later if we formalize menu orders.

File: guides/source/users/promotions/promotion-rules.html.md (hunk 3 above)
Comment (benjaminwil) [reply guides/source/users/promotions/promotion-rules.html.md]: Great edit!

File: guides/source/users/promotions/promotion-rules.html.md (hunk 4 above)
Comment (benjaminwil) [reply guides/source/users/promotions/promotion-rules.html.md]: I see what you mean!

**Comment by benjaminwil** [comment]
Responded and committed fixes for @aitbw's feedback. :+1:

File: guides/data/nav/users.yml (hunk 1 above)
Comment (kennyadsl) [reply guides/data/nav/users.yml]: I think it makes sense to read rules before actions since this is also the order we are presenting information in the UI.
